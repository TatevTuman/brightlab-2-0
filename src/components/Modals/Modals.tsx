import loadable from '@loadable/component'
import React, { memo } from 'react'

const HomeModal = loadable(() => import('./HomeModal/HomeModal'))
// here should be next modals

type Modal = typeof HomeModal // | here should be next modals
type Modals = Modal[]

// Modals
const modals: Modals = [
  HomeModal
  // here should be next modals
]

// Modals container
// It takes modalName and render necessary modal
const Modals = () => {
  return (
    <div id={'modals'}>
      {modals.map((Modal, index) => {
        return <Modal key={index} />
      })}
    </div>
  )
}

export default memo(Modals)
