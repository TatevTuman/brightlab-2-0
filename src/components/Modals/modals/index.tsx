import { withLoadableFallback } from '@hocs'
import type { HomeModalProps } from './HomeModal/HomeModal'

const HomeModal = withLoadableFallback<HomeModalProps>(import('./HomeModal/HomeModal'), {
  fallback: { height: '100vh' }
})

export { HomeModal }

export type { HomeModalProps }
