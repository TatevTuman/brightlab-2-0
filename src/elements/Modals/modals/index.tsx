import loadable from '@loadable/component'

import type { HomeModalProps } from './HomeModal/HomeModal'
const HomeModal = loadable(() => import('./HomeModal/HomeModal'))

export { HomeModal }

export type { HomeModalProps }
