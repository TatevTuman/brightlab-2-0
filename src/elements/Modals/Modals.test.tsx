import React from 'react'
import { render, cleanup, waitFor } from '@testing-library/react'
import Modals from './Modals'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('Modals', () => {
  it('renders correctly', async () => {
    const { container } = render(<Modals />)
    await waitFor(() => container)

    expect(container).toMatchSnapshot()
  })
})
