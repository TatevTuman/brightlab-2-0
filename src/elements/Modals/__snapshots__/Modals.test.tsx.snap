// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modals renders correctly 1`] = `
<div>
  <div
    id="modals"
  />
</div>
`;
